app.py

```diff
@@ -0,0 +1,138 @@
+import base64
+import json
+import logging
+import os
+import sys
+import boto3
+
+from chalice import Chalice
+
+app_name = 'FilterArcEventsApp'
+
+app = Chalice(app_name=app_name)
+
+# Initialize the logger
+logging.basicConfig(filename='app.log', filemode='w', format='%(name)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(app_name)
+
+# Set the log level from Environment variable LOG_LEVEL, by default INFO
+logger.setLevel(os.environ.get('LOG_LEVEL', logging.INFO))
+
+kinesis_endpoint_url = os.environ.get('KINESIS_ENDPOINT_URL', 'https://kinesis.us-east-1.amazonaws.com')
+aws_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', 'iam')
+aws_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', 'iam')
+aws_region = os.environ.get('AWS_REGION', 'us-east-1')
+output_stream_name = os.environ.get('OUTPUT_STREAM')
+logger.debug('##Environment variables {}'.format(os.environ))
+
+
+# function to create a client with aws for a specific service and region
+def create_client(service):
+    return boto3.client(
+        service,
+        endpoint_url=kinesis_endpoint_url,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+        region_name=aws_region)
+
+
+# function for sending data to Kinesis at the absolute maximum throughput
+def send_kinesis(kinesis_client, kinesis_stream_name, kinesis_record):
+    kinesisRecords = [kinesis_record]
+    # put the records to kinesis
+    kinesis_client.put_records(
+        Records=kinesisRecords,
+        StreamName=kinesis_stream_name
+    )
+
+
+def decode_record(record):
+    try:
+        logger.info("Decoded event with eventId {}".format(record['eventID']))
+        return base64.b64decode(record["kinesis"]["data"]).decode('utf-8')
+    except TypeError as ex:
+        logger.error("Type Error: {}".format(ex))
+
+
+def filter_event(record, filters):
+    """
+    A Generic method that removes the provided parameter from the given json record.
+    :param record: Kinesis record in json representation
+    :param filters: parameter in events that are required to be removed
+    :return: filtered json record
+    """
+    try:
+        for filter in filters:
+            # splitting with '.' as we can have nested parameters that needs to be filtered, like body.type
+            values = filter.split('.')
+            if len(values) > 1:
+                expr = 'del record'
+                for element_name in values:
+                    expr = expr + '[\'' + element_name + '\']'
+                code = compile(expr, 'delete_record', 'exec')
+                eval(code)
+            else:
```

Comment:
I think we will not need this condition now as even if it's a single element the above code should work, so we can remove the if and else condition

Reply:
ya it makes sense, will test and remove it